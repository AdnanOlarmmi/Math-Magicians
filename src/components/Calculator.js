/* eslint-disable */
import React, { useState } from 'react';
import calculate from '../logic/calculate';

const Calculator = () => {
  const [state, setState] = useState({ total: null, next: null, operation: null });

  const { total, next, operation } = state;

  const calculateValues = (e) => {
    const calcButton = e.target.innerHTML;
    const values = state;
    setState(calculate(values, calcButton));
  };

  return (
    <div className="calculator">
      <div className="output">
        { next || operation || total || 0 }
      </div>
      <table className="input" cellSpacing={0}>
        <tr>
          <td onClick={calculateValues}>AC</td>
          <td onClick={calculateValues}>+/-</td>
          <td onClick={calculateValues}>%</td>
          <td onClick={calculateValues}>รท</td>
        </tr>
        <tr>
          <td onClick={calculateValues}>7</td>
          <td onClick={calculateValues}>8</td>
          <td onClick={calculateValues}>9</td>
          <td onClick={calculateValues}>x</td>
        </tr>
        <tr>
          <td onClick={calculateValues}>4</td>
          <td onClick={calculateValues}>5</td>
          <td onClick={calculateValues}>6</td>
          <td onClick={calculateValues}>-</td>
        </tr>
        <tr>
          <td onClick={calculateValues}>1</td>
          <td onClick={calculateValues}>2</td>
          <td onClick={calculateValues}>3</td>
          <td onClick={calculateValues}>+</td>
        </tr>
        <tr>
          <td colSpan={2} onClick={calculateValues}>0</td>
          <td onClick={calculateValues}>.</td>
          <td onClick={calculateValues}>=</td>
        </tr>
      </table>
    </div>
  );
};

export default Calculator;
