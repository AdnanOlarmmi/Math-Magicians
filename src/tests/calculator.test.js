import { render } from "@testing-library/react";
import Calculator from "../components/Calculator";

describe("Confirming the display of calculator", () => {
  it("calculator displays", () => {
    const calculatorDisplay = render(<Calculator />);
    expect(calculatorDisplay).toMatchSnapshot();
  });
});
