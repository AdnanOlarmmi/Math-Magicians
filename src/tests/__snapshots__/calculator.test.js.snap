// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Confirming the display of calculator calculator displays 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <main
        class="calculator"
      >
        <h4>
          Let's do some Math
        </h4>
        <table
          cellspacing="0"
          class="input"
        >
          <tr>
            <td
              class="output"
              colspan="4"
            >
              0
            </td>
          </tr>
          <tr>
            <td>
              AC
            </td>
            <td>
              +/-
            </td>
            <td>
              %
            </td>
            <td>
              รท
            </td>
          </tr>
          <tr>
            <td>
              7
            </td>
            <td>
              8
            </td>
            <td>
              9
            </td>
            <td>
              x
            </td>
          </tr>
          <tr>
            <td>
              4
            </td>
            <td>
              5
            </td>
            <td>
              6
            </td>
            <td>
              -
            </td>
          </tr>
          <tr>
            <td>
              1
            </td>
            <td>
              2
            </td>
            <td>
              3
            </td>
            <td>
              +
            </td>
          </tr>
          <tr>
            <td
              colspan="2"
            >
              0
            </td>
            <td>
              .
            </td>
            <td>
              =
            </td>
          </tr>
        </table>
      </main>
    </div>
  </body>,
  "container": <div>
    <main
      class="calculator"
    >
      <h4>
        Let's do some Math
      </h4>
      <table
        cellspacing="0"
        class="input"
      >
        <tr>
          <td
            class="output"
            colspan="4"
          >
            0
          </td>
        </tr>
        <tr>
          <td>
            AC
          </td>
          <td>
            +/-
          </td>
          <td>
            %
          </td>
          <td>
            รท
          </td>
        </tr>
        <tr>
          <td>
            7
          </td>
          <td>
            8
          </td>
          <td>
            9
          </td>
          <td>
            x
          </td>
        </tr>
        <tr>
          <td>
            4
          </td>
          <td>
            5
          </td>
          <td>
            6
          </td>
          <td>
            -
          </td>
        </tr>
        <tr>
          <td>
            1
          </td>
          <td>
            2
          </td>
          <td>
            3
          </td>
          <td>
            +
          </td>
        </tr>
        <tr>
          <td
            colspan="2"
          >
            0
          </td>
          <td>
            .
          </td>
          <td>
            =
          </td>
        </tr>
      </table>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
